Userland/Libraries/LibCore/EventLoop.h

```diff
@@ -87,6 +87,27 @@ class EventLoop {
         post_event(context, make<Core::DeferredInvocationEvent>(context, move(invokee)));
     }
 
+    template<typename Result, typename ReturnValue = Optional<Result>>
+    ReturnValue try_callback(ErrorOr<Result> callback_value)
+    {
+        if (callback_value.is_error()) {
+            m_callback_error = callback_value.error();
+            return {};
+        }
+
+        return callback_value.release_value();
+    }
+
+    void try_callback(ErrorOr<void> callback_value)
+    {
+        if (callback_value.is_error()) {
+            m_callback_error = callback_value.error();
+            return;
+        }
+
+        return callback_value.release_value();
+    }
```

Comment:
This can probably just be
```c++
    void try_callback(ErrorOr<void> callback_value)
    {
        if (callback_value.is_error())
            m_callback_error = callback_value.error();
    }
```